mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java

```diff
@@ -0,0 +1,169 @@
+        super.onCreate(savedInstanceState);
+        ((MifosBaseActivity) getActivity()).getActivityComponent().inject(this);
+        noteItems = new ArrayList<>();
+        if (getArguments() != null) {
+            entityType = getArguments().getString(Constants.ENTITY_TYPE);
+            entityId = getArguments().getInt(Constants.ENTITY_ID);
+        }
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        noteItemPresenter.loadNote(entityType, entityId);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable
+            Bundle savedInstanceState) {
+        rootView = inflater.inflate(R.layout.fragment_notes, container, false);
+        ButterKnife.bind(this, rootView);
+        noteItemPresenter.attachView(this);
+
+        showUserInterface();
+        noteItemPresenter.loadNote(entityType, entityId);
+
+        return rootView;
+    }
+
+    @Override
+    public void onRefresh() {
+        noteItemPresenter.loadNote(entityType, entityId);
+    }
+
+    @Override
+    public void showUserInterface() {
+        setToolbarTitle(getResources().getString(R.string.note));
+        LinearLayoutManager mLayoutManager = new LinearLayoutManager(getActivity());
+        mLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
+        rvNote.setLayoutManager(mLayoutManager);
+        rvNote.setHasFixedSize(true);
+        rvNote.setAdapter(noteItemAdapter);
+        swipeRefreshLayout.setColorSchemeColors(getActivity()
+                .getResources().getIntArray(R.array.swipeRefreshColors));
+        swipeRefreshLayout.setOnRefreshListener(this);
+    }
+
+    @Override
+    public void showNoteItem(List<NoteItem> noteItems) {
+        this.noteItems = noteItems;
+        noteItemAdapter.setNotes(this.noteItems);
+    }
+
+    @Override
+    public void showEmptyNotes() {
+        ll_error.setVisibility(View.VISIBLE);
+        rvNote.setVisibility(View.GONE);
+        tvError.setText(R.string.empty_notes);
+    }
+
+    @Override
+    public void showResetVisibility() {
+        ll_error.setVisibility(View.GONE);
+        rvNote.setVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void showError(int message) {
+        Toaster.show(rootView, message);
+        ll_error.setVisibility(View.VISIBLE);
+        rvNote.setVisibility(View.GONE);
+        tvError.setText( getString(R.string.failed_to_fetch_notes));
+    }
+
+    @Override
+    public void showProgressbar(boolean show) {
+        if (show && (noteItemAdapter.getItemCount() == 0)) {
+            pbNote.setVisibility(View.VISIBLE);
+            swipeRefreshLayout.setRefreshing(false);
+        } else {
+            pbNote.setVisibility(View.GONE);
+            swipeRefreshLayout.setRefreshing(show);
+        }
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        noteItemPresenter.detachView();
```

Comment:
When you will add MifosBaseFragment Progressbar then do not forget to add `hideMifosProgressBar();` here.